Functions/QualityControl/xASL_qc_CleanOldQC.m

```diff
@@ -0,0 +1,94 @@
+function [x] = xASL_qc_CleanOldQC(x, ScanType, bRemoveCurrentSession)
+%xASL_qc_RemoveOutdatedQC Removes outdated QC parameters, loads QC from JSON and cleans the current session if necessary
+%
+% FORMAT: [x] = xASL_qc_RemoveOutdatedQC(x, ScanType[, bRemoveCurrentSession)
+%
+% INPUT:
+%   x 	                  - structure containing fields with all information required to run this submodule (REQUIRED)
+%   ScanType              - Type of the data - can be 'ASL' or 'func' (REQUIRED)
+%   bRemoveCurrentSession - Removes the current session completely (OPTIONAL, DEFAULT = false)
+%
+% OUTPUT:
+%   x        - same as input
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This function removes the old and outdated QC parameters. Specifically parameters for ASL and func which are now given in a subfield by session.
+%              It also loads further sessions from the disc and if necessary, it cleans the current session completely.
+%
+% EXAMPLE: x = xASL_qc_RemoveOutdatedQC(x, 'ASL');
+% __________________________________
+% Copyright (C) 2015-2023 ExploreASL	
+
+if nargin < 2
+	error('Two parameters are required');
+end
+
+if nargin < 3 || isempty(bRemoveCurrentSession)
+	bRemoveCurrentSession = false;
+end
+
+% All possible scantype lists
+ScanTypeList = {'func', 'ASL', 'dwi'};
```

Comment:
We need a warning here if the second input parameter doesn't agree with one of these here.

Reply:
Sorry. That's just a confusion. We do two things in this function.
1) We load all in QC.json to x-struct so that we can save QC.json later on while having all the QC for other sessions inside and not only for the current session.
2) We clean outdated tags of format x.ASL.parameter instead of x.ASL.ASL_1.parameter.

Point 1 needs to be done for all possible scantypes in QC.json. 
Point 2 was done only for the current scantype in input. But we can actually do it also for all scan types :)

So I'm removing the scantype from the input and adding more info and comments and redoing slightly the code.

Fixed in c523326a

Reply:
Nice!